core/drivers/crypto/caam/caam_ctrl.c

```diff
@@ -81,6 +73,20 @@ static TEE_Result crypto_driver_init(void)
 
 early_driver_init(crypto_driver_init);
 
+static TEE_Result crypto_driver_deinit(void)
```

Comment:
This is already managed by the original driver. The JR ring used during the initialization is released if the CFG_CRYPTO_DRIVER is not enabled, This function is not needed.

Reply:
The deinitialization is moved since I want the following sequence:
1. early_driver_init of the CAAM
2. service initcalls eventually call into the CAAM to request the MKVB
3. other service calls might use the CAAM to initialize RNG
3. driver_init_call late hands over all jobrings to normal world if `CFG_NXP_CAAM_RUNTIME_JR=n`

Reply:
RNG instantiation can not be done outside Secure environment. (it's may be working on some device not all).
You have to think that what is valid for i.MX6Q is not valid for i.MX7 or i.MX8. This is why this driver has been implemented in generic way with specific HAL.

Reply:
We are not leaving the secure environment, I don't understand how this moves the RNG initialization outside of the secure environment. Point 3. was meant that OP-TEE could use the CAAM to initialize its PRNG in case `CFG_NXP_CAAM_RUNTIME_JR=n`, I was not refering to the RNG instantiation of the RNG state handles inside the CAAM.